src/Database/PostgreSQL/Simple/Time/Internal/Parser.hs

```diff
@@ -193,3 +197,8 @@ zonedTime = Local.ZonedTime <$> localTime <*> (fromMaybe utc <$> timeZone)
 
 utc :: Local.TimeZone
 utc = Local.TimeZone 0 False ""
+
+calendarDiffTime :: Parser CalendarDiffTime
+calendarDiffTime = do
+  contents <- takeByteString
+  iso8601ParseM $ B8.unpack contents
```

Comment:
`Parser` is used here to leverage its instance of `MonadFail`, which `iso8601ParseM` needs. Ideally I wanted to use some `instance MonadFail (Either String)` instance, but apparently this deliberately doesn't exist in `base`.

See https://gitlab.haskell.org/ghc/ghc/-/issues/12160 for more.